From 5f66b57299d99cebf3ebdecbc2e343d7f0268be8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:57 -0400
Subject: [powerpc] eeh: Skip eeh sysfs when eeh is disabled

Message-id: <1410545655-205645-549-git-send-email-dzickus@redhat.com>
Patchwork-id: 94557
O-Subject: [RHEL7 PATCH 548/626] powerpc/eeh: Skip eeh sysfs when eeh is disabled
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2213fb142f1fff1ac28868d196294ca4423659af
Author: Wei Yang <weiyang@linux.vnet.ibm.com>
Date:   Wed Jun 4 09:49:17 2014 +0800

    powerpc/eeh: Skip eeh sysfs when eeh is disabled

    When eeh is not enabled, and hotplug two pci devices on the same bus, eeh
    related sysfs would be added twice for the first added pci device. Since the
    eeh_dev is not created when eeh is not enabled.

    This patch adds the check, if eeh is not enabled, eeh sysfs will not be
    created.

    After applying this patch, following warnings are reduced:

    sysfs: cannot create duplicate filename '/devices/pci0000:00/0000:00:00.0/eeh_mode'
    sysfs: cannot create duplicate filename '/devices/pci0000:00/0000:00:00.0/eeh_config_addr'
    sysfs: cannot create duplicate filename '/devices/pci0000:00/0000:00:00.0/eeh_pe_config_addr'

    Signed-off-by: Wei Yang <weiyang@linux.vnet.ibm.com>
    Acked-by: Gavin Shan <gwshan@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/eeh_sysfs.c b/arch/powerpc/kernel/eeh_sysfs.c
index 5d753d4..e2595ba 100644
--- a/arch/powerpc/kernel/eeh_sysfs.c
+++ b/arch/powerpc/kernel/eeh_sysfs.c
@@ -59,6 +59,9 @@ void eeh_sysfs_add_device(struct pci_dev *pdev)
  struct eeh_dev *edev = pci_dev_to_eeh_dev(pdev);
  int rc=0;
 
+ if (!eeh_enabled())
+  return;
+
  if (edev && (edev->mode & EEH_DEV_SYSFS))
   return;
 
-- 
1.7.1